### PR TITLE
Fix: Consistent SSL option for Download Clients

### PR DESCRIPTION
**Pull request by RobinDadswell**
#### Database Migration
YES

#### Description
To enable consistent ability to utilise SSL for all relevant download clients

#### Todos
- [ ] Tests
- [ ] Wiki Updates

#### Issues Fixed or Closed by this PR

* #4323

### CODE REVIEW

**Comment by RobinDadswell** [comment]
@bakerboy448  is there a wiki article to update for this?

**Comment by bakerboy448** [comment]
@RobinDadswell technically speaking no... but the download clients wiki entries could use some work... maybe at least a settings section that explains the options?

https://wiki.servarr.com/Settings_Download_Clients is the page to edit

**Comment by RobinDadswell** [comment]
> @RobinDadswell technically speaking no... but the download clients wiki entries could use some work... maybe at least a settings section that explains the options?
> 
> https://wiki.servarr.com/Settings_Download_Clients is the page to edit

I will have a look at it later; each client has slightly different settings though based on what it needs etc. so it may be a large Wiki piece - maybe not one for that page but as a subpage!

File: src/NzbDrone.Core/Download/Clients/NzbVortex/NzbVortexSettings.cs [hunk 1]
```diff
@@ -42,19 +43,22 @@ public NzbVortexSettings()
         [FieldDefinition(1, Label = "Port", Type = FieldType.Textbox)]
         public int Port { get; set; }
 
-        [FieldDefinition(2, Label = "Url Base", Type = FieldType.Textbox, Advanced = true, HelpText = "Adds a prefix to the NZBVortex url, e.g. http://[host]:[port]/[urlBase]/api")]
+        [FieldDefinition(2, Label = "Use SSL", Type = FieldType.Checkbox, HelpText = "Use secure connection when connecting to NZBVortex")]
```
Comment (markus101) [inline src/NzbDrone.Core/Download/Clients/NzbVortex/NzbVortexSettings.cs:46]: I think NZBVortex is the odd man out, where they only support SSL connections and using a reversing proxy to connect insecurely is indirectly. I think sticking with no setting makes sense.

File: src/NzbDrone.Core/Datastore/Migration/156_update_download_client_ssl_options.cs [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+using FluentMigrator;
+using Newtonsoft.Json.Linq;
+using NzbDrone.Common.Extensions;
+using NzbDrone.Common.Serializer;
+using NzbDrone.Core.Datastore.Migration.Framework;
+
+namespace NzbDrone.Core.Datastore.Migration
+{
+    [Migration(156)]
+    public class update_download_client_ssl_options : NzbDroneMigrationBase
```
Comment (markus101) [inline src/NzbDrone.Core/Datastore/Migration/156_update_download_client_ssl_options.cs:12]: I don't think this migration is necessary, it should inherit the default from settings when parsed from the DB.

**Review by markus101 — CHANGES_REQUESTED** [review]
I think NZBVortex should not have an option (they enforce SSL) and the migrations aren't necessary.

File: src/NzbDrone.Core/Datastore/Migration/156_update_download_client_ssl_options.cs (hunk 2 above)
Comment (RobinDadswell) [reply src/NzbDrone.Core/Datastore/Migration/156_update_download_client_ssl_options.cs]: removed

File: src/NzbDrone.Core/Download/Clients/NzbVortex/NzbVortexSettings.cs (hunk 1 above)
Comment (RobinDadswell) [reply src/NzbDrone.Core/Download/Clients/NzbVortex/NzbVortexSettings.cs]: Updated :)

**Comment by RobinDadswell** [comment]
Current failing test is not related to the changes made

**Review by markus101 — APPROVED** [review]

**Pull request merged**